GTG/gtk/ui/preferences.ui

```diff
@@ -0,0 +1,497 @@
+                        <property name="margin">20</property>
+                        <property name="margin-start">40</property>
+                        <child>
+                          <object class="GtkSwitch" id="bg_color_button">
+                            <property name="visible">True</property>
+                            <property name="valign">center</property>
+                            <signal handler="on_bg_color_toggled" name="notify::active" swapped="no"/>
+                          </object>
+                        </child>
+                        <child>
+                          <object class="GtkBox">
+                            <property name="orientation">vertical</property>
+                            <property name="valign">center</property>
+                            <property name="vexpand">false</property>
+                            <property name="visible">true</property>
+                            <property name="margin-start">10</property>
+                            <child>
+                              <object class="GtkLabel" id="bg_color_title">
+                                <property name="label">Enable coloured backgrounds</property>
+                                <property name="visible">true</property>
+                                <property name="xalign">0.0</property>
+                                <attributes>
+                                  <attribute name="weight" value="bold"/>
+                                </attributes>
+                              </object>
+                            </child>
+                            <child>
+                              <object class="GtkLabel" id="bg_color_label">
+                                <property name="label">If set, coloured backrouds will be enabled in the task list</property>
+                                <property name="visible">true</property>
+                                <property name="xalign">0.0</property>
+                                <style>
+                                  <class name="dim-label"/>
+                                </style>
+                                <attributes>
+                                  <attribute name="scale" value="0.833333"/>
+                                </attributes>
+                              </object>
+                            </child>
+                          </object>
+                        </child>
+                      </object>
+                    </child>
+                    <child>
+                      <object class="GtkBox" id="time_box">
+                        <property name="visible">True</property>
+                        <property name="can_focus">False</property>
+                        <property name="margin-bottom">20</property>
+                        <property name="margin-start">40</property>
+                        <child>
+                          <object class="GtkEntry" id="hour_entry">
```

Comment:
It might be worth to have just single entry instead and do correction from python. For example, when you enter `5pm`, it would automatically convert it to `17:00`. Have a play with Google Calendar that has a similar entry.